decidim-proposals/app/commands/decidim/proposals/admin/assign_proposals_to_valuator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Proposals
+    module Admin
+      # A command with all the business logic to assign proposals to a given
+      # valuator.
+      class AssignProposalsToValuator < Rectify::Command
+        # Public: Initializes the command.
+        #
+        # form - A form object with the params.
+        def initialize(form)
+          @form = form
+        end
+
+        # Executes the command. Broadcasts these events:
+        #
+        # - :ok when everything is valid.
+        # - :invalid if the form wasn't valid and we couldn't proceed.
+        #
+        # Returns nothing.
+        def call
+          return broadcast(:invalid) unless form.valid?
+
+          assignments = assign_proposals
+          return broadcast(:ok) if assignments.all?(&:persisted?)
```

Comment:
Mmm, I'm affraid that if any assignment is not persisted the transaction won't be rollbacked, but the command will return `:invalid`. Would be better to check if persisted inside the transaction?

Reply:
Can you check the last commit, see if that would fix it?

Reply:
Mmm, sorry, it still don't convince me, I guess that it won't work as the transaction wouldn't be rollbacked.

I've had similar problems with transactions and broadcasting messages and did something like this:
```ruby
transaction do
   # do things
   raise ActiveRecord::Rollback if failed?
   :ok
end | :invalid
```
Then you can broadcast the result of that block of code. But I don't know if there is any simpler way to do the same.

Reply:
See 15fd7ab, I added tests to ensure the code works. Thoughts about it? 😄

Reply:
Yeah, I like it that way! Sorry for being so nitpicky 😬

Reply:
No, no, you actually found a bug, it's good we have these reviews! 😄